whois-api/src/test/java/net/ripe/db/whois/api/nrtmv4/NrtmClientServiceTestIntegration.java

```diff
@@ -420,4 +478,20 @@ INSERT INTO version_info (id, source_id, version, session_id, type, last_serial_
         final String payloadNonAuth = "{\"nrtm_version\":4,\"timestamp\":\"2023-03-13T12:31:08Z\",\"type\":\"snapshot\",\"source\":\"TEST-NONAUTH\",\"session_id\":\"7f42be43-a1fd-48e7-947f-91c852cf13f6\",\"version\":1,\"snapshot\":{\"version\":1,\"url\":\"https://nrtm.ripe.net//7f42be43-a1fd-48e7-947f-91c852cf13f6/nrtm-snapshot.1.RIPE-NONAUTH.9a6ca0a3e1b68eb65d4a2d277c3e5a96.json.gz\",\"hash\":\"8140b2d3a3ec4cb6a14c1ca19f7cbca468adcc172ad3e313aed74bd0d1036838\"}}";
         databaseHelper.getNrtmTemplate().update("INSERT INTO notification_file (version_id, created, payload) VALUES (?, ?, ?)",2, dateTimeProvider.getCurrentDateTime().toEpochSecond(ZoneOffset.UTC), payloadNonAuth);
     }
+
+    public AsymmetricCipherKeyPair generateKeyPair() {
```

Comment:
Move this code into a separate DAO? Can we call it automatically at startup if nrtmv4 is enabled and no key pair is found already?

Reply:
Ok, in that case we should also call this method in case of generating new key pair while rotating? If its generated once a year I thought better to have it added manually.But I do not mind either way

Reply:
Yes also call this when rotating the key. We should automate it so we don't forget to do it manually